pkg/kubelet/server/remotecommand/websocket.go

```diff
@@ -17,61 +17,82 @@ limitations under the License.
 package remotecommand
 
 import (
+	"fmt"
 	"net/http"
 	"time"
 
 	"k8s.io/kubernetes/pkg/httplog"
+	"k8s.io/kubernetes/pkg/util/runtime"
 	"k8s.io/kubernetes/pkg/util/wsstream"
+)
 
-	"github.com/golang/glog"
+const (
+	stdinChannel = iota
+	stdoutChannel
+	stderrChannel
+	errorChannel
+	resizeChannel
 )
 
-// standardShellChannels returns the standard channel types for a shell connection (STDIN 0, STDOUT 1, STDERR 2)
-// along with the approximate duplex value. Supported subprotocols are "channel.k8s.io" and
-// "base64.channel.k8s.io".
-func standardShellChannels(stdin, stdout, stderr bool) []wsstream.ChannelType {
-	// open three half-duplex channels
-	channels := []wsstream.ChannelType{wsstream.ReadChannel, wsstream.WriteChannel, wsstream.WriteChannel}
-	if !stdin {
-		channels[0] = wsstream.IgnoreChannel
-	}
-	if !stdout {
-		channels[1] = wsstream.IgnoreChannel
+// createChannels returns the standard channel types for a shell connection (STDIN 0, STDOUT 1, STDERR 2)
```

Comment:
Ok, that's good enough for me.

On Wed, Jul 6, 2016 at 12:22 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/kubelet/server/remotecommand/websocket.go
> <https://github.com/kubernetes/kubernetes/pull/25273#discussion_r69763390>
> :
>
> >  )
> >
> > -// standardShellChannels returns the standard channel types for a shell connection (STDIN 0, STDOUT 1, STDERR 2)
> > -// along with the approximate duplex value. Supported subprotocols are "channel.k8s.io" and
> > -// "base64.channel.k8s.io".
> > -func standardShellChannels(stdin, stdout, stderr bool) []wsstream.ChannelType {
> > -	// open three half-duplex channels
> > -	channels := []wsstream.ChannelType{wsstream.ReadChannel, wsstream.WriteChannel, wsstream.WriteChannel}
> > -	if !stdin {
> > -		channels[0] = wsstream.IgnoreChannel
> > -	}
> > -	if !stdout {
> > -		channels[1] = wsstream.IgnoreChannel
> > +// createChannels returns the standard channel types for a shell connection (STDIN 0, STDOUT 1, STDERR 2)
>
> This only adds a new channel that new clients would be writing to. Old
> clients obviously won't have this channel and therefore won't write to it,
> so that's fine. I guess the question is what happens if a new client talks
> to an old server that doesn't recognize the resize channel. I think it'd be
> fine - it would hit this, right?
> https://github.com/ncdc/kubernetes/blob/exec-sigwinch/pkg/util/wsstream/conn.go#L233
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25273/files/a476748e44180a401eef8b62eda78f7f594a3fbd#r69763390>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p-kp1kXDXzBSeZ2uVbC9GC3mkWEEks5qS9ZRgaJpZM4IZGqq>
> .
>